### PR TITLE
MWPW-152375: Create new IMS client id and grant it permissions to Odin

### PR DESCRIPTION
**Pull request by vladen**
Uses new IMS client to authenticate currently logged in user in Odin.

IMS client: https://imss.corp.adobe.com/#/client/prod/mas-studio
Resolves https://jira.corp.adobe.com/browse/MWPW-152375

Test URLs:
- Before: https://main--mas--adobecom.hlx.live/studio.html
- After: https://MWPW-152375--mas--adobecom.hlx.page/studio.html

Note: IMS portal does not allow default redirect URL to include double dash (--) that is common in Milo URLs. That's why the default redirect URL for IMS client is configured as https://www.adobe.com/mas/studio.html

image

### CODE REVIEW

**Review by yesil — DISMISSED** [review]

File: studio.html [hunk 1]
```diff
@@ -1,26 +1,66 @@
...
+                locale: 'en_US',
+                redirect_uri: location.href.substring(
+                    0,
+                    location.href.length - location.hash.length,
+                ),
+                response_type: 'id_token',
+                scope: 'additional_info,additional_info.projectedProductContext,additional_info.roles,AdobeID,openid,read_organizations,read_pc,read_pc.acp,read_pc.dma_aem_ams,read_pc.dma_aem_cloud,read_pc.dma_tartan',
+                useLocalStorage: true,
+                onAccessToken(result) {
+                    if (!localStorage.getItem('masAccessToken')) {
```
Comment (yesil) [inline studio.html:36]: We probably don't need localStorage once it works via IMS.
we should use IMS api to retrieve the token in studio and use it.

Comment (vladen) [reply studio.html]: Done

**Review by Axelcureno — APPROVED** [review]
🚀 Thanks @vladen !

File: commons/src/aem.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import { Fragment } from '../../studio/src/store/Fragment.js';
 
-const accessToken = localStorage.getItem('masAccessToken');
+const accessToken = adobeIMS.getAccessToken().token;
```
[the remark this answers is not in the document]
Comment (vladen) [reply commons/src/aem.js:3]: @yesil the method seems to be sync:
image

[the remark this answers is not in the document]
Comment (yesil) [reply commons/src/aem.js]: awesome thanks

**Review by yesil — APPROVED** [review]

**Pull request merged**